### ISSUE TITLE
Allow conditional logic in terms definitions

### ISSUE THREAD

**Issue by bwiernik** [citation-style-language/schema#282]
**Is your feature request related to a problem? Please describe.**

Some languages have multiple forms for a term depending on the item type (usually books versus journals). For example, [Turkish](https://forums.zotero.org/discussion/81900/is-it-possible-different-ibid-term-for-books-article-and-web-page#latest) uses different phrases for "ibid" for books vs. journals. [German](https://github.com/citation-style-language/csl-evolution/issues/34) uses different words for "volume" for books vs. journals. 

**Describe the solution you'd like**
@denismaier suggested we allow `choose` within terms: https://github.com/citation-style-language/csl-evolution/issues/34#issuecomment-624647159. I think that will work to accommodate this.

**Comment by denismaier** [comment]
That would be a useful feature. I think this should be added. I'll add this to the 1.1 milestone for now. (We can always defer if it's not feasible currently.)

@bdarcus Opinions?

**Comment by denismaier** [comment]
I will prepare a PR for this? Or may there be simpler solutions for this?
@bdarcus @bwiernik ?

**Comment by bdarcus** [comment]
Seems sensible.

BTW, while it adds an extra step, if you're so inclined, it would be a good idea when we do these PRs to also support one or more tests to the "1.1-experimental" branch of the test-suite?

**Comment by denismaier** [comment]
Hmm. I'm not sure we can just allow `choose` in this context, as `cs:choose` has `rendering-element*` as child elements. I don't think we will want to change the normal pattern for `cs:choose` right? 
So this means we need a new pattern for `choose` for terms?

**Comment by bdarcus** [comment]
Right.

**Comment by denismaier** [comment]
Then, at what level will we want to add it? Under each term?
This is the example in the specs:
```xml
<terms>
    <term name="no date">n.d.</term>
    <term name="et-al">et al.</term>
    <term name="page">
      <single>page</single>
      <multiple>pages</multiple>
    </term>
    <term name="page" form="short">
      <single>p.</single>
      <multiple>pp.</multiple>
    </term>
  </terms>
```

We could allow this:

```xml
<terms>
    <term name="no date">
      <choose>
        <if type="article">n.d.
        </if>
        <else>asdf
        </else>
       </choose>
     </term>
    <term name="et-al">et al.</term>
    <term name="page">
      <single>page</single>
      <multiple>pages</multiple>
    </term>
    <term name="page" form="short">
      <choose>
        <if type="article">
          <single>p.</single>
          <multiple>pp.</multiple>
        </if>
        <else>
          <single>p.</single>
          <multiple>pp.</multiple>
        </else>
       </choose>
    </term>
  </terms>
```

**Comment by denismaier** [comment]
Or: We could actually just do this:
```xml
<terms>
    <term name="volume" form="short" type="article-journal">
      <single>vol.</single>
      <multiple>vols.</multiple>
    </term>
    <term name="volume" form="short" type="book">
      <single>something else</single>
      <multiple>something else</multiple>
    </term>
  </terms>
```

**Comment by bwiernik** [comment]
Just item type matching would need to able to match="any" against multiple types (eg, `type="book chapter"`). It would also likely need a way to specify `else` to indicate all other types (e.g., maybe one term for article-journal, article-magazine, article-newspaper, and a different term for all other types).

**Comment by denismaier** [comment]
> It would also likely need a way to specify else to indicate all other types (e.g., maybe one term for article-journal, article-magazine, article-newspaper, and a different term for all other types).

Right. We could say that elment order matters:

```xml
<terms>
    <term name="volume" form="short">
      <single>vol.</single>
      <multiple>vols.</multiple>
    </term>
    <term name="volume" form="short" type="book">
      <single>something else</single>
      <multiple>something else</multiple>
    </term>
  </terms>
```

**Comment by bwiernik** [comment]
If we did that, it should be the other way around--first specific types, then the final for all remaining.

I wonder if it would be clearer, though, to just add that `cs:if`, `cs:else-if`, and `cs:else` may be children to `cs:term` and give them their specific behavior in that context (in the same way that `cs:label` has different behavior when a child of `cs:names` versus otherwise.

**Comment by denismaier** [comment]
> I wonder if it would be clearer, though, to just add that `cs:if`, `cs:else-if`, and `cs:else` may be children to `cs:term` and give them their specific behavior in that context (in the same way that `cs:label` has different behavior when a child of `cs:names` versus otherwise.

So, basically something like this:
```terms.term =
element cs:term {
  term.attributes, (terms.term.elem | terms.term.choose)
}

terms.term.elem = text | (term.single, term.multiple)

terms.term.choose =
  
  ## Use to conditionally render rendering elements.
  element cs:choose { choose.if, choose.else-if*, choose.else? }

## When specifying conditions, we allow either the attribute syntax or the element-based one.
choose.condition =
  (condition.atts, match?)
  | element cs:conditions { match, terms.term.elem }
choose.if = element cs:if { choose.condition, terms.term.elem }
choose.else-if =
  element cs:else-if { choose.condition, terms.term.elem }
choose.else = element cs:else { terms.term.elem }
```

What do you think?

**Comment by bwiernik** [comment]
I think that looks good. Let's be sure it tests correctly.

**Comment by denismaier** [comment]
I'm not sure: Do we need to change the names of the patterns? So `terms.term.choose.condition` instead of `choose.condition`?

**Comment by bwiernik** [comment]
That's a question for @bdarcus

**Comment by denismaier** [comment]
Also: Which test should we allow here? All tests that we currently have with `cs: choose`? Or `@type`?

**Comment by denismaier** [comment]
Closed by #308
